forced_photometry/code_src/galex_functions.py

```diff
@@ -0,0 +1,178 @@
+    verbose : bool, optional
+        Whether to print status messages. Default is True.
+
+    Returns
+    -------
+    downloaded_files : list of str
+        List of paths to the downloaded files.
+    """
+
+    if verbose:
+        print(f"Querying GALEX around RA={coords.ra.deg:.5f}, Dec={coords.dec.deg:.5f}...")
+
+    Observations.enable_cloud_dataset()
+    obs_table = Observations.query_criteria(
+        coordinates=coords,
+        radius=search_radius_arcsec * u.arcsec,
+        obs_collection="GALEX"
+    )
+
+    if len(obs_table) == 0:
+        print("No GALEX observations found for the target.")
+        return []
+
+    if verbose:
+        print(f"Found {len(obs_table)} observations. Getting product lists...")
+
+    product_tables = [Observations.get_product_list(obs) for obs in obs_table]
+    all_products = vstack(product_tables)
+
+
+    # Filter for SCIENCE + skybg files
+    is_sci = (all_products['productType'] == 'SCIENCE') & (all_products['dataproduct_type'] == 'image')
+    desc_col = all_products['description']
+    desc_filled = np.array([str(x) if x is not np.ma.masked else "" for x in desc_col])
+    is_bkg = np.char.find(desc_filled, 'skybg') >= 0
+    product_subset = all_products[is_sci | is_bkg]
+
+        # Count and list how many match
+    print(f"Found {np.sum(is_bkg)} files with 'skybg' in the filename")
+
+    
+    # Filter by exposure time
+    obs_ids_good = set(obs_table[obs_table['t_exptime'] > min_exptime]['obs_id'])
+    is_good_obs = np.array([obs_id in obs_ids_good for obs_id in product_subset['obs_id']])
+
+    final_products = product_subset[is_good_obs]
+
+    # Filter out unwanted file types like -cat.fits.gz or -xd-mcat.fits.gz
+    unwanted_substrings = ["-cat.fits", "-xd-mcat.fits"]
+    final_products = final_products[
+        [not any(substr in name for substr in unwanted_substrings)
+         for name in final_products['productFilename']]
+    ]
+
+   
+    # Download files
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    manifest = Observations.download_products(final_products, 
+                                              mrp_only=False, 
+                                              download_dir=output_dir)
+    print(manifest.colnames)
+    downloaded_files = []
+    for observation in manifest:
+
+        #get the path to the downloaded file
+        local_path = observation['Local Path']
+        fname = os.path.basename(local_path)
+
+        #put these files in the output directory
+        #not the 'mastDownloads' directory structure
+        final_path = os.path.join(output_dir,fname)
+        shutil.move(local_path, final_path)
+        downloaded_files.append(final_path)
+
+    if verbose:
+        print("Download complete.")
+
+    return downloaded_files
+
+def galex_get_skybg(coords: SkyCoord,
+                    output_dir: str = "data/Galex",
+                    verbose: bool = True) -> list[str]:
```

Comment:
I don't think we want to complicate this with type hinting (you can leave it, but I would prefer to understand the 1) need for it 2) do we plan to enforce it?